manifests/client.pp

```diff
@@ -9,27 +27,142 @@
+    $server_list = $server_random_order_based_on_fqdn.join(' ')
+
+    # Local variables
+    $sshdir = '/root/.ssh'
+
+    $file_defaults = {
+      ensure  => file,
+      owner   => root,
+      group   => root,
+      mode    => '0600',
+      require => File[$sshdir],
+    }
+
+    # Define unique parameters of each resource
+    $data = {
+      $work_directory => {
+        ensure => directory,
+        mode   => '0700',
+        require => [],
+      },
+      $sshdir => {
+        ensure => directory,
+        mode   => '0700',
+        require => [],
+      },
+      "${sshdir}/backup_id_${ssh_key_type}" => {
+        content => Sensitive($ssh_key_priv),
+      },
+      "${sshdir}/backup_id_${ssh_key_type}.pub" => {
+        content => Sensitive("${ssh_key_type} ${ssh_key_pub} backup@${facts['networking']['fqdn']}\n"),
+        mode    => '0644',
+      },
+    }
+
+    # Ensure the resources
+    ensure_resources( 'file', $data, $file_defaults )
+
+    # Ensure borg_defaults file and settings
+    # CONVERT $prune_settings HASH TO $prune_options_string
+    $prune_options_string = $prune_settings
+    .map | $key, $value | { "--${key} ${value}" }
+    .join(' ')
+    file { "${work_directory}/borg_defaults.sh":
+      ensure  => file,
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0700',
+      content => template('profile_backup/borg_defaults.sh.erb'),
+      require => File[$work_directory],
+    }
+
+    file { "${work_directory}/borg_check.sh":
+      ensure  => file,
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0700',
+      content => template('profile_backup/borg_check.sh.erb'),
+      require => File[$work_directory],
+    }
+
+    file { "${work_directory}/borg_list.sh":
+      ensure  => file,
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0700',
+      content => template('profile_backup/borg_list.sh.erb'),
+      require => File[$work_directory],
+    }
+
+    file { "${work_directory}/borg_backup_job.sh":
+      ensure  => file,
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0700',
+      content => template('profile_backup/borg_backup_job.sh.erb'),
+      require => File[$work_directory],
+    }
+
+    # EXPORT CLIENT DETAILS TO BACKUP SERVER FOR ACCESS
+    @@profile_backup::server::allow_client { "allow host ${facts['networking']['fqdn']} access to backup servers":
+      hostname     => $facts['networking']['fqdn'],
+      ip           => $facts['networking']['ip'],
+      ssh_key_pub  => $ssh_key_pub,
+      ssh_key_type => $ssh_key_type,
+      tag          => 'profile_backup_allow_client',
+    }
+
+    $cron_ensure = 'present'
+  } else {
+    $cron_ensure = 'absent'
+  }
+
+  # BACKUP CRONS
+
+  cron { 'profile_backup verify':
+    ensure  => $cron_ensure,
+    command => "${work_directory}/borg_check.sh 2>&1 | logger -s -t backup &>/dev/null",
```

Comment:
Perhaps use a unique syslog tag for the _backup check_ logs to make them easier to identify.